frontend/src/components/Network.tsx

```diff
@@ -423,6 +428,147 @@ const Network = ({ nodes: _nodes, edges: _edges }: Props) => {
     [_edges, nodes, minEdgeStrength, maxEdgeStrength, edgeColors],
   );
 
+  const setupCytoscapePanLimits = () => {
+    let panning = false;
+    let currentPan: Pan | undefined = undefined;
```

Comment:
`Pan` here could be something like `ReturnType<typeof cy.pan>`, or the Cytoscape types might just export a `Pan` type that you can just import (they probably/hopefully do).